node/src/accountant/payment_adjuster/preparatory_analyser/mod.rs

```diff
@@ -332,33 +321,36 @@ impl ServiceFeeSingleTXErrorFactory<ServiceFeeImmoderateInsufficiency>
 #[derive(Debug, PartialEq, Eq, Clone)]
 pub struct LateServiceFeeSingleTxErrorFactory {
     original_number_of_accounts: usize,
-    original_service_fee_required_total_minor: u128,
+    original_total_service_fee_required_minor: u128,
 }
 
 impl LateServiceFeeSingleTxErrorFactory {
-    pub fn new(unadjusted_accounts: &[WeightedPayable]) -> Self {
+    pub fn new(unadjusted_accounts: &[WeighedPayable]) -> Self {
         let original_number_of_accounts = unadjusted_accounts.len();
-        let original_service_fee_required_total_minor =
-            sum_as(unadjusted_accounts, |account| account.balance_minor());
+        let original_total_service_fee_required_minor = sum_as(unadjusted_accounts, |account| {
+            account.initial_balance_minor()
+        });
         Self {
             original_number_of_accounts,
-            original_service_fee_required_total_minor,
+            original_total_service_fee_required_minor,
         }
     }
 }
 
-impl ServiceFeeSingleTXErrorFactory<PaymentAdjusterError> for LateServiceFeeSingleTxErrorFactory {
+impl ServiceFeeSingleTXErrorFactory<PaymentAdjusterError, WeighedPayable>
+    for LateServiceFeeSingleTxErrorFactory
+{
     fn make(
         &self,
-        number_of_accounts: usize,
-        _required_service_fee_total: u128,
+        current_set_of_accounts: &[WeighedPayable],
         cw_service_fee_balance_minor: u128,
     ) -> PaymentAdjusterError {
-        PaymentAdjusterError::LateNotEnoughFeeForSingleTransaction {
+        let number_of_accounts = current_set_of_accounts.len();
+        PaymentAdjusterError::AbsolutelyInsufficientServiceFeeBalancePostTxFeeAdjustment {
```

Comment:
Please consider shortening this error name:

1. `ServiceFeeAbsoluteInsufficiencyPostTxFeeAdjustment`
2. `InadequateServiceFee`